boards: Add nrf53 bsim

auroraslb opened the pull request:
This PR adds support for simulating the NRF53 network core. To be able to use bluetooth, you will need https://github.com/zephyrproject-rtos/zephyr/pull/30732.

Note: Bluetooth does not fully work, will need some help debugging this.

auroraslb committed:
nrf5x_bsim: Rename board folder nrf52_bsim -> nrf5x_bsim

This folder will contain common content for both nrf52_bsim
and nrf53_cpunet_bsim.

Signed-off-by: Aurora Sletnes Bjørlo <aurora.sletnes@nordicsemi.no>

auroraslb committed:
nrf5x_bsim: Add the board nrf53_cpunet_bsim

The board models the nrf53840 cpunet.

Signed-off-by: Aurora Sletnes Bjørlo <aurora.sletnes@nordicsemi.no>

auroraslb committed:
nrf5x_bsim: Add irq logging for nrf53

Ensure correct logging of irq with nrf53.

Signed-off-by: Aurora Sletnes Bjørlo <aurora.sletnes@nordicsemi.no>

auroraslb committed:
nrf5x_bsim: Add defines for COMPATIBLE_NRF53X_CPUNET

Select COMPATIBLE_NRF53X_CPUNET in SOC_NRF5340_CPUNET.
Also add defined(CONFIG_SOC_SERIES_NRF53X) where we
have defined(NRF53_SERIES).

Signed-off-by: Aurora Sletnes Bjørlo <aurora.sletnes@nordicsemi.no>

auroraslb committed:
west.yml: Update nrf_hw_models version to include nrf53 support

Update nrf_hw_models version to include nrf53 support.

Signed-off-by: Aurora Sletnes Bjørlo <aurora.sletnes@nordicsemi.no>

rugeGerritsen commented on boards/posix/nrf5x_bsim/irq_handler.c:
```diff
@@ -29,6 +29,42 @@ static struct _isr_list irq_vector_table[NRF_HW_NBR_IRQs];
+#if defined(CONFIG_SOC_SERIES_NRF53X) || \
+    defined(CONFIG_SOC_COMPATIBLE_NRF53X_CPUNET) || \
+    defined(NRF53_SERIES)
+const char *irqnames[] = { /*just for the traces*/
```
You may consider adding the remaining IRQ names as well, see nrf5340_network.h. 
* IPC
* EGU

rugeGerritsen reviewed (commented):
You simplify some conditionals similar to:
`defined(CONFIG_SOC_SERIES_NRF53X) || defined(CONFIG_SOC_COMPATIBLE_NRF53X_CPUNET)`
by adding `select SOC_COMPATIBLE_NRF53X_CPUNET` to `SOC_NRF5340_CPUNET` in`zephyr/soc/arm/nordic_nrf/nrf53/Kconfig.soc`

auroraslb committed:
CODEOWNERS: rename nrf52_bsim to nrf5x_bsim

Rename nrf52_bsim to nrf5x_bsim to match with change due to
added nrf53_bsim.

Signed-off-by: Aurora Sletnes Bjørlo <aurora.sletnes@nordicsemi.no>

galak commented:
Is the intent to replace nrf52_bsim?

rugeGerritsen commented:
> Is the intent to replace nrf52_bsim?

@galak. I would assume nrf53_bsim can be used as an addition to nrf52_bsim. The HAL is slightly different between nrf52 and nrf53 series devices. @cvinayak and @carlescufi  may comment.

galak commented:
> @galak. I would assume nrf53_bsim can be used as an addition to nrf52_bsim. The HAL is slightly different between nrf52 and nrf53 series devices. @cvinayak and @carlescufi may comment.

it would be good to explain what the difference / value is between the two.  This is more about a question from a testing/ci point of view.

auroraslb committed:
modules: CMakeLists: Add definition for CONFIG_SOC_COMPATIBLE_NRF5340_CPUNET

So that the nrfx drivers can be compiled for a simulated
nrf53 cpunet target.

Signed-off-by: Aurora Sletnes Bjørlo <aurora.sletnes@nordicsemi.no>

auroraslb committed:
nrf5x_bsim: Add and choose SOC_COMPATIBLE_BSIM

Add to the SOC_COMPATIBLE options a common option to choose
compatibility with a simulated BabbleSim target. Also choose
this option for the simulated targets nrf52_bsim and nrf53_cpunet_bsim.

Signed-off-by: Aurora Sletnes Bjørlo <aurora.sletnes@nordicsemi.no>
